### ISSUE TITLE
Flask sessions and cookies for multiple users

### ISSUE THREAD

**Issue by aditirao7** [marshalmiller/rottingresearch#7]
Enable proper usage for multiple users using flask sessions and cookies. 
Currently, 2 different users analysing the same pdf and downloading references raises an error because files already exist in the downloads folder.

**Comment by aditirao7** [comment]
@marshalmiller was wondering whether having user authentication and storing all the analyzed PDF reports so the user can view them later could be a potential feature? I've noticed that most PDF related tools do this. Also do you know the reach of this webpage currently?

**Comment by marshalmiller** [comment]
@aditirao7 It might be, but it would be far down the line. The issue is with the collection of user data. We'd have to update GDPR and a bunch of other stuff. We do have partnerships and funding from places like Docker, RedHat, and The Open Source Collective. So we have to abide by their guidelines. They are on board for us collecting stats but not user-identifiable information. The saving of PDFs is similarly grey, as technically, we are distributing copies of the files—and storage costs. Right now, they just get deleted right after. Once the database is set up to collect the stats, I'm not against returning to them and seeing what we can do about logins. However, I think the main interest is researching the data collected about the links themselves.

As far as stats, that one is a bit more complicated. We have about 5k downloads of the docker image. The Python library behind it also has about 1k active users. The live website hovers around 300 unique visitors a month. Of course, until now, it had very little information on it. I'm at a conference next week and plan on talking about the project at another one in January. So, I anticipate the live site getting more traffic soon.

**Comment by aditirao7** [comment]
@marshalmiller Makes sense, thanks for the detailed response. I'm going to try working on the database. Also can I email you? I want to know more about the conferences, sounds exciting!

**Comment by marshalmiller** [comment]
@aditirao7 I just sent you an email using the email on your website. Feel free to reach out any time.
